packages/block-library/src/navigation-overlay-close/edit.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * WordPress dependencies
+ */
+import { Button, Icon, ToggleControl, PanelBody } from '@wordpress/components';
+import { close } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+import { useBlockProps, InspectorControls } from '@wordpress/block-editor';
+import { privateApis as routerPrivateApis } from '@wordpress/router';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../lock-unlock';
+
+const { useHistory } = unlock( routerPrivateApis );
+
+export default function Edit( { attributes, isSelected, setAttributes } ) {
+	const blockProps = useBlockProps();
+	const history = useHistory();
+	const { hasIcon } = attributes;
+
+	const closeText = __( 'Close' );
+
+	const onClick = () => {
+		if ( isSelected ) {
+			// Exit navigation overlay edit mode.
+			history.back();
```

Comment:
There are more routes available, so this doesn't work. Example repro:


- From the Site Editor
- Go to the Overlay Template Editor
- Click the Site Logo/W in the top Left to open up the left Sidebar
- Click the Editor Canvas
- Click the Overlay X
- The Editor Sidebar will open, and you can't exit the overlay edit directly back to the edtior

Reply:
Also, we should still have the Back button navigation in the central header area as on other templates.

Reply:
> Also, we should still have the Back button navigation in the central header area as on other templates.

That's turned into a wider problem. I agree we need it but it will need to be a seperate PR.